validate/validate.go

```diff
@@ -294,10 +294,18 @@ func (v *Validator) CheckMounts() (msgs []string) {
 		return
 	}
 
-	if supportedTypes != nil {
-		for _, mount := range v.spec.Mounts {
-			if !supportedTypes[mount.Type] {
-				msgs = append(msgs, fmt.Sprintf("Unsupported mount type %q", mount.Type))
+	for i, fMount := range v.spec.Mounts {
+		if supportedTypes != nil && !supportedTypes[fMount.Type] {
+			msgs = append(msgs, fmt.Sprintf("Unsupported mount type %q", fMount.Type))
+		}
+		for j, bMount := range v.spec.Mounts {
+			if nestedValid(bMount.Destination, fMount.Destination) {
+				if v.spec.Platform.OS == "windows" {
+					msgs = append(msgs, fmt.Sprintf("On Windows, %v nested within %v is forbidden", bMount.Destination, fMount.Destination))
```

Comment:
Looking over the spec language again, I'm not clear that it's forbidding covering mounts (e.g. mounting `c:\foo` *after* mounting `c:\foo\bar`).  If that's the intention, I think we need stronger/clearer spec language before forbidding them here (vs. just warning about them, like you're doing for Linux).